pkg/bootstrap/docker/host/host.go

```diff
@@ -76,155 +72,45 @@ func (h *HostHelper) EnsureVolumeShare() error {
-		OutputStream: localTarFile,
-	})
-	if err != nil {
-		return err
-	}
-	if err = localTarFile.Close(); err != nil {
-		return err
-	}
-	localTarClosed = true
-	inputTar, err := os.Open(localTarFile.Name())
-	if err != nil {
-		return err
-	}
-	defer func() {
-		errors.LogError(inputTar.Close())
-	}()
-	tarHelper := tarhelper.New()
-	tarHelper.SetExclusionPattern(nil)
-	glog.V(4).Infof("Extracting temporary tar %s to directory %s", inputTar.Name(), destDir)
-	var tarLog io.Writer
-	if glog.V(5) {
-		tarLog = os.Stderr
-	}
-	return tarHelper.ExtractTarStreamWithLogging(destDir, inputTar, tarLog)
-}
-
-// makeTempCopy creates a temporary directory and places a copy of the source file
-// in it. It returns the directory where the temporary copy was made.
-func makeTempCopy(file string) (string, error) {
-	tempDir, err := ioutil.TempDir("", "")
-	if err != nil {
-		return "", err
-	}
-	destPath := filepath.Join(tempDir, filepath.Base(file))
-	destFile, err := os.Create(destPath)
-	if err != nil {
-		return "", err
-	}
-	defer func() {
-		errors.LogError(destFile.Close())
-	}()
-	sourceFile, err := os.Open(file)
+	err = dockerhelper.DownloadDirFromContainer(h.client, container, sourceDir, destDir)
 	if err != nil {
-		return "", err
+		glog.V(4).Infof("An error occurred downloading the directory: %v", err)
+	} else {
+		glog.V(4).Infof("Successfully downloaded directory.")
 	}
-	defer func() {
-		errors.LogError(sourceFile.Close())
-	}()
-	_, err = io.Copy(destFile, sourceFile)
-	return tempDir, err
+	return err
 }
 
-// CopyMasterConfigToHost copies a local file to the Docker host
-func (h *HostHelper) CopyMasterConfigToHost(sourceFile, destDir string) error {
-	localDir, err := makeTempCopy(sourceFile)
-	if err != nil {
-		return err
-	}
-	defer os.RemoveAll(localDir)
-	tarHelper := tarhelper.New()
-	tarHelper.SetExclusionPattern(nil)
-	var tarLog io.Writer
-	if glog.V(5) {
-		tarLog = os.Stderr
-	}
-	localTarFile, err := ioutil.TempFile("", "master-config")
-	if err != nil {
-		return err
-	}
-	localTarClosed := false
-	defer os.Remove(localTarFile.Name())
-	defer func() {
-		if !localTarClosed {
-			errors.LogError(localTarFile.Close())
-		}
-	}()
-	glog.V(4).Infof("Creating temporary tar %s to upload to %s", localTarFile.Name(), destDir)
-	err = tarHelper.CreateTarStreamWithLogging(localDir, false, localTarFile, tarLog)
-	if err != nil {
-		return err
-	}
-	err = localTarFile.Close()
-	if err != nil {
-		return err
-	}
-	localTarClosed = true
-	localTarInputClosed := false
-	localTarInput, err := os.Open(localTarFile.Name())
-	if err != nil {
-		return err
-	}
-	defer func() {
-		if !localTarInputClosed {
-			localTarInput.Close()
-		}
-	}()
-	bind := fmt.Sprintf("%s:/var/lib/origin/openshift.local.config:z", destDir)
+// UploadFileToContainer copies a local file to the Docker host
+func (h *HostHelper) UploadFileToContainer(src, dst string) error {
+	bind := fmt.Sprintf("%s:/var/lib/origin/openshift.local.config:z", h.configDir)
```

Comment:
Same here, if we're always mounting the local config, then the function name should reflect that.